### ISSUE TITLE
Timeout in System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http3.ConnectionAttemptCanceled_AuthorityNotBlocked

### ISSUE THREAD

**Issue by lewing** [dotnet/runtime#100971]
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=640906
Build error leg or test failing: System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http3.ConnectionAttemptCanceled_AuthorityNotBlocked
Pull request: https://github.com/dotnet/runtime/pull/100520

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

```json
{
  "ErrorMessage": " at System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http3.ConnectionAttemptCanceled_AuthorityNotBlocked",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=640906
**Error message validated:** `[ at System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http3.ConnectionAttemptCanceled_AuthorityNotBlocked`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 4/12/2024 5:07:56 PM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[654644](https://dev.azure.com/dnceng-public/public/_build/results?buildId=654644)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_AltSvc_Test_Http3.AltSvc_Header_Upgrade_Success](https://dev.azure.com/dnceng-public/public/_build/results?buildId=654644&view=ms.vss-test-web.build-test-results-tab&runId=16200474&resultId=214800)|dotnet/runtime#101447|
|[653932](https://dev.azure.com/dnceng-public/public/_build/results?buildId=653932)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_AltSvc_Test_Http3.AltSvc_ConnectionFrame_UpgradeFrom20_Success](https://dev.azure.com/dnceng-public/public/_build/results?buildId=653932&view=ms.vss-test-web.build-test-results-tab&runId=16179954&resultId=215673)|dotnet/runtime#101449|
|[653556](https://dev.azure.com/dnceng-public/public/_build/results?buildId=653556)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_HttpClientHandlerTest_Headers_Http3.SendAsync_Expires_Success](https://dev.azure.com/dnceng-public/public/_build/results?buildId=653556&view=ms.vss-test-web.build-test-results-tab&runId=16169518&resultId=215601)|dotnet/runtime#101435|
|[653075](https://dev.azure.com/dnceng-public/public/_build/results?buildId=653075)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_HttpClientHandlerTest_Headers_Http3.SendAsync_Expires_Success](https://dev.azure.com/dnceng-public/public/_build/results?buildId=653075&view=ms.vss-test-web.build-test-results-tab&runId=16152630&resultId=215577)|dotnet/runtime#101418|
|[652395](https://dev.azure.com/dnceng-public/public/_build/results?buildId=652395)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_HttpClientHandlerTest_Headers_Http3.SendAsync_Expires_Success](https://dev.azure.com/dnceng-public/public/_build/results?buildId=652395&view=ms.vss-test-web.build-test-results-tab&runId=16135426&resultId=215374)|dotnet/runtime#101396|
|[652158](https://dev.azure.com/dnceng-public/public/_build/results?buildId=652158)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_HttpClientHandlerTest_Http3.ReadAsStreamAsync_HandlerProducesWellBehavedResponseStream](https://dev.azure.com/dnceng-public/public/_build/results?buildId=652158&view=ms.vss-test-web.build-test-results-tab&runId=16128656&resultId=214448)|dotnet/runtime#101389|
|[651133](https://dev.azure.com/dnceng-public/public/_build/results?buildId=651133)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Cookies_Http3.GetAsync_SetCookieContainerAndCookieHeader_BothCookiesSent](https://dev.azure.com/dnceng-public/public/_build/results?buildId=651133&view=ms.vss-test-web.build-test-results-tab&runId=16099268&resultId=215079)|dotnet/runtime#101330|
|[650694](https://dev.azure.com/dnceng-public/public/_build/results?buildId=650694)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_AltSvc_Test_Http3.AltSvc_ConnectionFrame_UpgradeFrom20_Success](https://dev.azure.com/dnceng-public/public/_build/results?buildId=650694&view=ms.vss-test-web.build-test-results-tab&runId=16087188&resultId=215815)|dotnet/runtime#101336|
|[649002](https://dev.azure.com/dnceng-public/public/_build/results?buildId=649002)|dotnet/runtime|[System.Net.Http.Functional.Tests.HttpMetricsTest_Http11_Async_HttpMessageInvoker.RequestDuration_ConnectionClosedWhileReceivingHeaders_Recorded](https://dev.azure.com/dnceng-public/public/_build/results?buildId=649002&view=ms.vss-test-web.build-test-results-tab&runId=16042810&resultId=214696)|dotnet/runtime#101279|
|[647960](https://dev.azure.com/dnceng-public/public/_build/results?buildId=647960)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_HttpClientHandlerTest_Headers_Http3.SendAsync_Expires_Success](https://dev.azure.com/dnceng-public/public/_build/results?buildId=647960&view=ms.vss-test-web.build-test-results-tab&runId=16011492&resultId=215359)|dotnet/runtime#101226|
|[647349](https://dev.azure.com/dnceng-public/public/_build/results?buildId=647349)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_HttpClientHandlerTest_Http3.ReadAsStreamAsync_HandlerProducesWellBehavedResponseStream](https://dev.azure.com/dnceng-public/public/_build/results?buildId=647349&view=ms.vss-test-web.build-test-results-tab&runId=15990230&resultId=214544)|dotnet/runtime#80154|
|[647034](https://dev.azure.com/dnceng-public/public/_build/results?buildId=647034)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_HttpClientHandlerTest_Headers_Http3.SendAsync_Expires_Success](https://dev.azure.com/dnceng-public/public/_build/results?buildId=647034&view=ms.vss-test-web.build-test-results-tab&runId=15978222&resultId=215417)|dotnet/runtime#101192|
|[645041](https://dev.azure.com/dnceng-public/public/_build/results?buildId=645041)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_MaxResponseHeadersLength_Http3.LargeSingleHeader_ThrowsException](https://dev.azure.com/dnceng-public/public/_build/results?buildId=645041&view=ms.vss-test-web.build-test-results-tab&runId=15917522&resultId=217842)|dotnet/runtime#100951|
|[644210](https://dev.azure.com/dnceng-public/public/_build/results?buildId=644210)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_MaxResponseHeadersLength_Http3.LargeSingleHeader_ThrowsException](https://dev.azure.com/dnceng-public/public/_build/results?buildId=644210&view=ms.vss-test-web.build-test-results-tab&runId=15895506&resultId=217644)|dotnet/runtime#100951|
|[641013](https://dev.azure.com/dnceng-public/public/_build/results?buildId=641013)|dotnet/runtime|[System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http3.ConnectionAttemptCanceled_AuthorityNotBlocked](https://dev.azure.com/dnceng-public/public/_build/results?buildId=641013&view=ms.vss-test-web.build-test-results-tab&runId=15806338&resultId=218842)|dotnet/runtime#100951|
|[640906](https://dev.azure.com/dnceng-public/public/_build/results?buildId=640906)|dotnet/runtime|[System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http3.ConnectionAttemptCanceled_AuthorityNotBlocked](https://dev.azure.com/dnceng-public/public/_build/results?buildId=640906&view=ms.vss-test-web.build-test-results-tab&runId=15804076&resultId=215452)|dotnet/runtime#100520|
|[640844](https://dev.azure.com/dnceng-public/public/_build/results?buildId=640844)|dotnet/runtime|[System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http3.ConnectionAttemptCanceled_AuthorityNotBlocked](https://dev.azure.com/dnceng-public/public/_build/results?buildId=640844&view=ms.vss-test-web.build-test-results-tab&runId=15800484&resultId=215452)|dotnet/runtime#100951|
|[640726](https://dev.azure.com/dnceng-public/public/_build/results?buildId=640726)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_MaxResponseHeadersLength_Http3.LargeSingleHeader_ThrowsException](https://dev.azure.com/dnceng-public/public/_build/results?buildId=640726&view=ms.vss-test-web.build-test-results-tab&runId=15796286&resultId=214106)|dotnet/runtime#100669|
|[639824](https://dev.azure.com/dnceng-public/public/_build/results?buildId=639824)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_MaxResponseHeadersLength_Http3.LargeSingleHeader_ThrowsException](https://dev.azure.com/dnceng-public/public/_build/results?buildId=639824&view=ms.vss-test-web.build-test-results-tab&runId=15770208&resultId=214417)|dotnet/runtime#100767|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|1|10|19|

**Comment by ManickaP** [comment]
https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-100520-merge-ab726ead71c24b32bb/System.Net.Http.Functional.Tests/3/console.69d8426b.log?helixlogtype=result

mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.16-helix-amd64 
```
System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http3.ConnectionAttemptCanceled_AuthorityNotBlocked [FAIL]
      System.TimeoutException : The operation has timed out.
      Stack Trace:
        /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs(55,0): at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks, Int32 millisecondsTimeout)
        /_/src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs(1142,0): at System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http3.ConnectionAttemptCanceled_AuthorityNotBlocked()
        --- End of stack trace from previous location ---
```

**Comment by karelz** [comment]
Fixed in main (9.0) in PR #101250 and in 8.0.7 in PR #102147 (July release).
